moby/moby
# daemon: allow setting OCI annotations on containers

- Relates to #45030
- Closes #39142
- Closes #37289
- Closes #36181
- Relates to #37262



**- What I did**

Allow API clients to set and retrieve OCI annotations on a container.

**- How I did it**

Added a field to `HostConfig` and plumbed it into the OCI spec generation code.

**- How to verify it**
Create a container with annotations and start it. Inspect the running container using `ctr` and verify that the annotations are listed in the container spec.

A new integration test verifies that annotations can be round-tripped through the Engine API.

**- Description for the changelog**

- Annotations—arbitrary non-identifying metadata—can now be attached to containers on creation which will be passed to the runtime when the container is started.

**- A picture of a cute animal (not mandatory but encouraged)**
